src/_pytest/terminal.py

```diff
@@ -726,11 +727,19 @@ def summary_warnings(self):
             if not all_warnings:
                 return
 
+            final = hasattr(self, "_already_displayed_warnings")
```

Comment:
Hmm I'm not sure, I understand the rationale for the change, but I'm not so happy with how this turns out... 🤔